mobile/src/main/java/org/openhab/habdroid/ui/widget/PeriodicSignalImageButton.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.ui.widget
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import androidx.annotation.CallSuper
+import androidx.appcompat.widget.AppCompatImageButton
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class PeriodicSignalImageButton constructor(context: Context, attrs: AttributeSet?) :
+    AppCompatImageButton(context, attrs),
+    View.OnLongClickListener,
+    View.OnTouchListener {
+    private var scope: CoroutineScope? = null
+    private var periodicCallbackExecutor: Job? = null
+
+    var callback: ((v: View, value: String?) -> Unit)? = null
+    var clickCommand: String? = null
+    var longClickHoldCommand: String? = null
+
+    init {
+        setOnLongClickListener(this)
+        setOnTouchListener(this)
+    }
+
+    @CallSuper
+    override fun onLongClick(v: View?): Boolean {
+        scheduleNextSignal()
+        return true
+    }
+
+    @CallSuper
```

Comment:
While this annotation seems correct, shouldn't you follow it by calling the super method below?

Reply:
From the java docs: "Denotes that any overriding methods should invoke this method as well."
I added the annotation here, so the methods overriding it need to call super, but not this one.

Reply:
Oh, I see now ... I didn't realize that this view is a touch listener for itself. Could probably also override `onTouchEvent` and call the super method instead of using the listener?

Reply:
Done